client/app/src/js/app.js

```diff
@@ -134,6 +143,16 @@ function AttachTokens($window) {
   };
 }
 
+function run($rootScope, $location, Auth) {
```

Comment:
This stuff is actually genuinely mysterious, even to me. (I need to remind myself how it works.) Some notes here would be useful to everyone that touches it later.